src/main/java/kitchenpos/menu/application/MenuService.java

```diff
@@ -0,0 +1,60 @@
+package kitchenpos.menu.application;
+
+import kitchenpos.menu.domain.Menu;
+import kitchenpos.menu.domain.MenuRepository;
+import kitchenpos.menu.dto.MenuRequest;
+import kitchenpos.menu.dto.MenuResponse;
+import kitchenpos.menu.group.domain.MenuGroup;
+import kitchenpos.menu.group.domain.MenuGroupRepository;
+import kitchenpos.product.domain.Product;
+import kitchenpos.product.domain.ProductRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.*;
+
+@Service
+@Transactional(readOnly = true)
+public class MenuService {
+    private final MenuRepository menuRepository;
+    private final MenuGroupRepository menuGroupRepository;
+    private final ProductRepository productRepository;
+
+    public MenuService(
+            final MenuRepository menuRepository,
+            final MenuGroupRepository menuGroupRepository,
+            final ProductRepository productRepository
+    ) {
+        this.menuRepository = menuRepository;
+        this.menuGroupRepository = menuGroupRepository;
+        this.productRepository = productRepository;
+    }
+
+    @Transactional
+    public MenuResponse create(final MenuRequest menuRequest) {
+
+        final MenuGroup menuGroup = menuGroupRepository.findById(menuRequest.getMenuGroupId())
+                .orElseThrow(() -> new EntityNotFoundException("MenuGroup이 존재하지 않습니다. MenuGroupId = " + menuRequest.getMenuGroupId()));
+
+        final Menu menu = menuRequest.toEntity();
+        menu.grouping(menuGroup);
+
+        final List<Long> productIds = menuRequest.getProductIds();
+        final List<Product> products = productRepository.findAllById(productIds);
+
+        if(products.size() != productIds.size()) {
```

Comment:
```suggestion
        if (products.size() != productIds.size()) {
```

`Product `와 관련된 부분들은 `ProductService`로 위임해보시는건 어떨까요??
현재 `MenuService`에서 연관을 가지고 있는 `Repository`는 총 3개 입니다 

```java
private final MenuRepository menuRepository;
private final MenuGroupRepository menuGroupRepository;
private final ProductRepository productRepository;

```

만약 `Menu`와 관련된 부분만 따로 도메인을 분리 한다거나 이 부분이 `MSA`로 별도의 서비스로 구성 된다면
이 부분은 의존성이 발생하는 부분입니다 
(혹 `Product`와 관련된 비즈니스가 수정되었을 경우 이 부분도 함께 수정되어야 합니다)
`Product`부분은 `Product`와 관련된 도메인 혹은 서비스로 위임하는게 더 자연스러울 것 같습니다 😄 

이 부분은 한번 고민 부탁 드리겠습니다 🙇

Reply:
네 말씀하신대로 Menu -> Product를 의존하고 있어 Product가 변경될 경우 수정될 부분이 많을 것 같습니다. 
Product 관련된 서비스에 위임하는 방식이 좋을 것 같습니다